### PR TITLE
drivers: wifi: simplelink: reimplement getaddrinfo

### LINKED ISSUE #11890: Reimplement getaddrinfo() to call SlNetUtil_getaddrinfo() in new SimpleLink SDK v 2.30+
The current implementation of `getaddrinfo()` for the SimpleLink WiFi offload driver only handles client mode.  
Once issue #11126 is complete to update SimpleLink SDK to ver 2.30, then `getaddrinfo()` should be updated to call the new utility function `SlNetUtil_getaddrinfo()`, which handles both client and server mode, and includes better error checking.

### PR DESCRIPTION
**Pull request by vanti**
Reimplement getaddrinfo to call SlNetUtil_getAddrInfo from the TI HAL
for a more robust implementation that supports both client and server
modes, and performs better error-checking.

Fixes #11890

~~Marked DNM til https://github.com/zephyrproject-rtos/hal_ti/pull/20 and https://github.com/zephyrproject-rtos/zephyr/pull/27590 are merged.~~

### CODE REVIEW

**Review by pfalcon — APPROVED** [review]

File: drivers/wifi/simplelink/simplelink_sockets.c [hunk 1]
```diff
@@ -974,64 +973,15 @@ static ssize_t simplelink_sendmsg(void *obj, const struct msghdr *msg,
...
- * Later SimpleLink SDK versions implement the full getaddrinfo semantics,
- * returning potentially multiple IP addresses.
- * This version implements a simple gethostbyname() API for client only.
- */
-static int simplelink_getaddrinfo(const char *node, const char *service,
-				  const struct zsock_addrinfo *hints,
-				  struct zsock_addrinfo **res)
+/* Adds address info entry to a list */
+static int set_addr_info(const struct SlNetUtil_addrInfo_t *sl_ai,
+	struct zsock_addrinfo **res)
```
Comment (jukkar) [inline drivers/wifi/simplelink/simplelink_sockets.c:978]: Indentation

File: drivers/wifi/simplelink/simplelink_support.c [hunk 2]
```diff
@@ -65,6 +71,16 @@ struct sl_connect_state sl_conn;
 /* Network Coprocessor state, including role and connection state: */
 static struct nwp_status nwp;
 
+/* Configuration of SlNetIfWifi for Zephyr */
+static SlNetIf_Config_t slnetifwifi_config_zephyr = {
+	SlNetIfWifi_socket, SlNetIfWifi_close,
```
[the remark this answers is not in the document]
Comment (vanti) [reply drivers/wifi/simplelink/simplelink_support.c]: Done

File: drivers/wifi/simplelink/simplelink_sockets.c (hunk 1 above)
Comment (vanti) [reply drivers/wifi/simplelink/simplelink_sockets.c]: Done

**Comment by vanti** [comment]
Thanks @jukkar. I made the changes based on your comments.

**Review by jukkar — APPROVED** [review]

**Review by tbursztyka — APPROVED** [review]

**Comment by vanti** [comment]
Rebased due to #24873.

**Pull request merged**